include/CL/sycl/vec/detail/vec.hpp

```diff
@@ -90,8 +94,45 @@ class vec : public detail::small_array<DataType,
     return std::tuple_cat(flatten<V>(i)...);
   }
 
+protected:
+  __swizzled_base_vec__<DataType, 1>do_swizzle(int first) const {
+    return (*this)[first];
+  }
+
+  template<typename T, typename... Ts>
+  __swizzled_base_vec__<DataType, sizeof...(Ts) + 1>do_swizzle(T first, Ts... swizzleIndexes) const {
+    return __swizzled_base_vec__<DataType, sizeof...(Ts) + 1> {do_swizzle(first), do_swizzle(swizzleIndexes...) };
```

Comment:
Now we have interpolation on variadic template parameter packs, I have the feeling it should be possible to avoid recursion and to write just a
```
template<typename T, typename... Ts>
  __swizzled_base_vec__<DataType, sizeof...(Ts)>do_swizzle(Ts... swizzleIndexes) const {
    return { (*this)[swizzleIndexes]...) };
```
or perhaps with `((*this)[swizzleIndexes])...`
There are a few examples on https://en.cppreference.com/w/cpp/language/parameter_pack

Reply:
" error: expected binary operator before ‘)’ token"

I'll try and dig a bit more about possible answers, the parameter pack stuff is quite confusing to get correct syntax for.

Reply:
I'm not finding antyhing that works for me here, happy to throw any ideas at it, but this is definitely bumping up against my C++ knowledge edges.